modules/Bio/EnsEMBL/Variation/Pipeline/PhenotypeAnnotation/BasePhenotypeAnnotation.pm

```diff
@@ -0,0 +1,970 @@
+    LIMIT 1
+  };
+
+  my $pf_ins_stmt = qq{
+    INSERT INTO phenotype_feature (
+      phenotype_id, source_id, study_id,
+      type, object_id, is_significant,
+      seq_region_id, seq_region_start, 
+      seq_region_end, seq_region_strand
+    )
+    VALUES (
+      ?, ?, ?,
+      ?, ?, ?,
+      ?, ?, 
+      ?, ?
+    )
+  };
+
+  my $attrib_ins_stmt = qq{
+    INSERT INTO phenotype_feature_attrib (
+      phenotype_feature_id,
+      attrib_type_id,
+      value
+    )
+    SELECT ?, at.attrib_type_id, ?
+    FROM attrib_type at
+    WHERE at.code = ?
+  };
+
+  my $attrib_ins_cast_stmt = qq{
+    INSERT INTO phenotype_feature_attrib (
+      phenotype_feature_id,
+      attrib_type_id,
+      value
+    )
+    SELECT ?, at.attrib_type_id, CAST(? AS CHAR)
+    FROM attrib_type at
+    WHERE at.code = ?
+  };
+
+  my $attrib_id_ext_stmt = qq{
+    SELECT attrib_id, value from attrib, attrib_type
+    WHERE attrib_type.code = 'ontology_mapping'
+    AND attrib.attrib_type_id = attrib_type.attrib_type_id
+  };
+
+  my $ontology_accession_ins_stmt = qq{
+    INSERT IGNORE INTO phenotype_ontology_accession
+    (phenotype_id, accession, mapped_by_attrib, mapping_type)
+    values (?,?,?,?)
+   };
+
+  my $st_ins_sth   = $db_adaptor->dbc->prepare($st_ins_stmt);
+  my $pf_check_sth = $db_adaptor->dbc->prepare($pf_check_stmt);
+  my $pf_ins_sth   = $db_adaptor->dbc->prepare($pf_ins_stmt);
+  my $attrib_ins_sth = $db_adaptor->dbc->prepare($attrib_ins_stmt);
+  my $attrib_ins_cast_sth = $db_adaptor->dbc->prepare($attrib_ins_cast_stmt);
+  my $ontology_accession_ins_sth = $db_adaptor->dbc->prepare($ontology_accession_ins_stmt);
+
+  ## get the attrib id for the type of description to ontology term linking
+  my $attrib_id_ext_sth = $db_adaptor->dbc->prepare($attrib_id_ext_stmt);
+  $attrib_id_ext_sth->execute();
+  my $ont_attrib_type = $attrib_id_ext_sth->fetchall_hashref("value");
+
+  if ($source_info->{source_name} eq 'RGD'){ #required as db source:name is 'RGD' and attrib_type_id 509 for ontology_mapping is 'Rat Genome Database'
+    $source_info->{source_attrib_type} = 'Rat Genome Database';
+  }
+  my $mapped_by = 'Data source';
+  if (exists $source_info->{source_mapped_attrib_type} && exists $ont_attrib_type->{$source_info->{source_mapped_attrib_type}}){
+    $mapped_by = $source_info->{source_mapped_attrib_type};
+  }
+
+  # First, sort the array according to the phenotype description
+  my @sorted = sort {($a->{description} || $a->{name}) cmp ($b->{description} || $b->{name})} @{$phenotypes};
+  my $study_count = 0;
+  my $phenotype_feature_count = 0;
+
+  my $total = scalar @sorted;
+  my $i = 0;
+
+  while (my $phenotype = shift(@sorted)) {
+    progress($i++, $total);
+
+    $object_type = $phenotype->{type} if defined($phenotype->{type});
+
+    my $object_id = $phenotype->{"id"};
+    # If the rs could not be mapped to a variation id, skip it
+    next if $object_type =~ /Variation/ && (!defined($variation_ids->{$object_id}));
```

Comment:
We need to report what is being skipped here.

Reply:
This is not reported in log file: WARN: skipping rsID: could not be mapped to a variation id : $object_id